src/SFA.DAS.ProviderApprenticeshipsService.Web/Authorization/AuthorizationContextProvider.cs

```diff
@@ -54,7 +55,12 @@ public IAuthorizationContext GetAuthorizationContext()
 
         private long? GetProviderId(RouteValueDictionary routeValueDictionary)
         {
-            if (long.TryParse((string) routeValueDictionary[RouteDataKeys.ProviderId], out var providerId))
+            long providerId;
+
+            if (long.TryParse(_httpContext.User.Identity.GetClaim("http://schemas.portal.com/ukprn"), out providerId))
+                return providerId;
+
+            if (long.TryParse((string) routeValueDictionary[RouteDataKeys.ProviderId], out providerId))
                 return providerId;
```

Comment:
It is always a worry that there are no tests for much of the web layer (but this isn't an issue created here)

Reply:
There _are_ tests for this class.